skimage/segmentation/morphsnakes.py

```diff
@@ -0,0 +1,440 @@
+    if len(image.shape) not in [2, 3]:
+        raise ValueError("Input image is not a 2D or 3D array.")
+
+    if len(image.shape) != len(init_level_set.shape):
+        raise ValueError("The dimensions of the initial level set do not "
+                         "match the dimensions of the image.")
+
+
+def _find_threshold(image, percentile=20):
+    """Find a proper threshold for `morph_gac` using the given percentile."""
+    return np.percentile(image, percentile)
+
+
+def _init_level_set(init_level_set, image_shape):
+    """Auxiliary function for initializing level sets with a string.
+
+    If `init_level_set` is not a string, it is returned as is.
+    """
+    if isinstance(init_level_set, str):
+        if init_level_set == 'checkerboard':
+            res = checkerboard_level_set(image_shape)
+        elif init_level_set == 'circle':
+            res = circle_level_set(image_shape)
+        else:
+            raise ValueError("`init_level_set` not in "
+                             "['checkerboard', 'circle']")
+    else:
+        res = init_level_set
+    return res
+
+
+def circle_level_set(image_shape, center=None, radius=None):
+    """Create a circle level set with binary values.
+
+    Parameters
+    ----------
+    image_shape : tuple of positive integers
+        Shape of the image
+    center : tuple of positive integers, optional
+        Coordinates of the center of the circle given in (row, column). If not
+        given, it defaults to the center of the image.
+    radius : float, optional
+        Radius of the circle. If not given, it is set to the 75% of the
+        smallest image dimension.
+
+    Returns
+    -------
+    out : array with shape `image_shape`
+        Binary level set of the circle with the given `radius` and `center`.
+
+    See also
+    --------
+    checkerboard_level_set
+    """
+
+    if center is None:
+        center = tuple(i // 2 for i in image_shape)
+
+    if radius is None:
+        radius = min(image_shape) * 3.0 / 8.0
+
+    grid = np.mgrid[[slice(i) for i in image_shape]]
+    grid = (grid.T - center).T
+    phi = radius - np.sqrt(np.sum((grid)**2, 0))
+    res = np.int8(phi > 0)
+    return res
+
+
+def checkerboard_level_set(image_shape, square_size=5):
+    """Create a checkerboard level set with binary values.
+
+    Parameters
+    ----------
+    image_shape : tuple of positive integers
+        Shape of the image.
+    square_size : int, optional
+        Size of the squares of the checkerboard. It defaults to 5.
+
+    Returns
+    -------
+    out : array with shape `image_shape`
+        Binary level set of the checkerboard.
+
+    See also
+    --------
+    circle_level_set
+    """
+
+    grid = np.mgrid[[slice(i) for i in image_shape]]
+    grid = (grid // square_size) & 1
+    checkerboard = np.bitwise_xor.reduce(grid, axis=0)
+    res = np.int8(checkerboard)
+    return res
+
+
+def gborders(image, alpha=100.0, sigma=5.0):
```

Comment:
This function name is somewhat unintuitive---do you have any other suggestions?

Also, the one line description "Stopping criterion for image borders." does not make its function immediately clear to me.

Reply:
I agree. How about `inverse_gaussian_gradient`?

Reply:
I like `inverse_gaussian_gradient`. I'll use that.